selfdrive/manager.py

```diff
@@ -97,20 +109,29 @@ def unblock_stdout():
         line = line.rstrip()
         prefix = b'progress: '
         if line.startswith(prefix):
-          i = int(line[len(prefix):])
+          progress = 70.0 * (int(line[len(prefix):]) / TOTAL_SCONS_NODES)
           if spinner is not None:
-            spinner.update("%d" % (50.0 * (i / TOTAL_SCONS_NODES)))
-        elif len(line):
-          print(line.decode('utf8'))
+            spinner.update("%d" % progress)
+
+        if len(line) and not line.startswith(prefix):
+          line = line.decode('utf8')
+          print(line)
+          if any([err in line for err in ['error: ', 'not found, needed by target']]):
+            build_error = True
+            spinner.update("%d" % progress, format_spinner_error(line))
+            time.sleep(10)
+            break
```

Comment:
You should also wait for the scons command to finish. Otherwise you might start the clean command while it's still building. Because it's building in multiple threads an error can show up, and it could still be finishing things in another thread.